### ISSUE TITLE
ERR_PNPM_TARBALL_EXTRACT in version 8.7.0

### ISSUE THREAD

**Issue by AminFazlMondo** [pnpm/pnpm#6991]
### Verify latest release

- [ ] I verified that the issue exists in the latest pnpm release

### pnpm version

8.7.0

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Reproduction steps

We have a private scoped package from AWS CodeArtifact that I can install with `8.6.x` but when it is in GitHub Action and using the lastest version 8, It errors.

### Describe the Bug

Below is the message that we get in installing the dependencies step:

```
WARN  GET https://<ourOrg>-artifacts-***.d.codeartifact.ap-southeast-2.amazonaws.com/npm/***/***/***/-/***-2.92.4.tgz error (ERR_PNPM_TARBALL_EXTRACT). Will retry in 10 seconds. 2 retries left.
```
This will retry couple of times and then exits

### Expected Behavior

Should install the package successfully.

### Which Node.js version are you using?

18

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [X] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

Ubuntu, GitHub Actions

**Comment by oculus42** [comment]
Related to #6761 ?

**Comment by zkochan** [comment]
No, not related to that issue. See my comment https://github.com/pnpm/pnpm/issues/6761#issuecomment-1696618705

**Comment by zkochan** [comment]
:ship: 8.7.1

**Comment by zkochan** [comment]
I don't know
does it work with v8.6?

**Comment by zkochan** [comment]
no, you issue has nothing to do with this because your issue is in 7.33.6

**Comment by Fish1** [comment]
I am running into this issue with 8.9 for https://registry.npmjs.org/@devcontainers/cli/-/cli-0.52.0.tgz 

```
 WARN  GET https://registry.npmjs.org/@devcontainers/cli/-/cli-0.52.0.tgz error (ERR_PNPM_TARBALL_EXTRACT). Will retry in 10 seconds. 2 retries left.
 WARN  GET https://registry.npmjs.org/@devcontainers/cli/-/cli-0.52.0.tgz error (ERR_PNPM_TARBALL_EXTRACT). Will retry in 1 minute. 1 retries left.
```

**Comment by qingchuwudi** [comment]
I meet the same problem, and resolved with the command:
```bash
## prune old packages 
pnpm store prune
## find the path  and reomve it.
rm -rf `pnpm store path`
## reinstall
pnpm install
```

**Comment by karkir0003** [comment]
getting same issue with 8.15

**Comment by e-mannonen** [comment]
same problem

**Comment by miripiruni** [comment]
> … works with sudo tho

I think it's just retry for flaky behaviour…

**Comment by VladimirNT** [comment]
Please recheck in v9.12.3 with [the fix merged in](https://github.com/VladimirNT/pnpm/commit/6b633d43a6f282f72fffd76f330aba916587b615) for similar problem
